### ISSUE TITLE
Container deletion on scanner node is not working properly

### ISSUE THREAD

**Issue by dharmit** [CentOS/container-pipeline-service#504]
In prod, the scanner node is getting filled up with numerous containers in `Exited` or `Created` state that show up when you do a `docker ps -a`. What's happening is that we end up with a file system like this:

```bash
$ ll /
total 62
drwxr-xr-x.  17 root root   272 Mar  2 20:17 0a0ff0cc646871c8ae5f487ea91435eb42159a1fdaa1bfeeb7d48246319e35be
drwxr-xr-x.  17 root root   272 Mar  2 22:43 27b5b66fcc11bf06baf20682abfdfe451cff5f14ff5a36ce9ae2258b9953fa0b
drwxr-xr-x.  17 root root   272 Mar  3 09:35 3c401e3e6ff947986491429601a66562cc1fa39b5a7eea5dcc974eabbdc765af
drwxr-xr-x.  17 root root   272 Mar  3 18:34 8e002deab476694b384ee6c4ba993fd69806782defc7ca5bdf28860d7b342aeb
drwxr-xr-x.  21 root root   252 Mar  4 01:55 9416b028c154ab713e62383b20001ed82db7fe3e48fd3c59d4ef248ec72e187d
drwxr-xr-x.  17 root root   267 Mar  4 04:38 b06be640bcea06f8571fdafda69403dea4964873e6e406522c9475e3bac0368a
lrwxrwxrwx.   1 root root     7 Oct  5 11:17 bin -> usr/bin
dr-xr-xr-x.   6 root root  1024 Jan 17 05:48 boot
drwxr-xr-x.  19 root root  3340 Mar  5 02:57 dev
drwxr-xr-x.  91 root root  4096 Mar  1 09:21 etc
drwxr-xr-x.  17 root root   272 Mar  2 11:11 fe8d512e5b81abce6011c0454b8b96f94a1f015e2af270f9248b54fface7c267
drwxr-xr-x.  11 root root  4096 Oct  5 11:33 home
lrwxrwxrwx.   1 root root     7 Oct  5 11:17 lib -> usr/lib
lrwxrwxrwx.   1 root root     9 Oct  5 11:17 lib64 -> usr/lib64
drwx------.   2 root root 16384 Oct  5 11:17 lost+found
drwxr-xr-x.   2 root root  4096 Nov  5  2016 media
drwxr-xr-x.   2 root root  4096 Nov  5  2016 mnt
drwxr-xr-x.   3 root root  4096 Oct 24 08:50 opt
drwxr-xr-x.   3 root root  4096 Oct 28 04:41 ostree
dr-xr-xr-x. 217 root root     0 Jan 17 12:19 proc
dr-xr-x---.   4 root root  4096 Dec  7 13:52 root
drwxr-xr-x.  31 root root  1000 Mar  1 09:22 run
lrwxrwxrwx.   1 root root     8 Oct  5 11:17 sbin -> usr/sbin
drwxr-xr-x.   3 root root  4096 Oct 18 06:16 srv
dr-xr-xr-x.  13 root root     0 Jan 17 12:19 sys
drwxrwxrwt.  11 root root  4096 Mar  4 04:27 tmp
drwxr-xr-x.  13 root root  4096 Oct  5 11:17 usr
drwxr-xr-x.  19 root root  4096 Oct  5 11:24 var
```

The directories named after SHA of an image in above output contain the container image's rootfs and this causes the `/` to get filled up fast. We end up with filling up entire disk space on the scanner node.

So, besides the great work in https://github.com/CentOS/container-pipeline-service/pull/486 and https://github.com/CentOS/container-pipeline-service/pull/452, we need to make more changes to ensure that scanner node doesn't succumb to disk space issues.

**Comment by dharmit** [comment]
What I have observed is that removing these containers in `Created`/`Exited` space frees up a few GBs of disk space without removing any mounts from `/` in above output.

**Comment by navidshaikh** [comment]
> removing these containers in Created/Exited space frees up a few GBs of disk space 

I think one of the reason is due to the scenario that we had docker thinpool configured earlier, later fixed by docker-storage-setup.

I think there are two problems here
1 - The containers in dangling state
2-  The atomic scan mount and failure to unmount upon completing execution.

**Comment by navidshaikh** [comment]
fix coming in with #558
